internal/batch/batch_processor.go

```diff
@@ -277,6 +282,29 @@ func (bp *batchProcessor) captureFlushError(err error) {
 	fs.LastFlushError = err.Error()
 }
 
+func (bp *batchProcessor) cancelFlush(ctx context.Context, id *fftypes.UUID) error {
+	bp.statusMux.Lock()
+	defer bp.statusMux.Unlock()
+	fs := &bp.flushStatus
+
+	if bp.conf.txType != core.TransactionTypeContractInvokePin {
+		return i18n.NewError(ctx, coremsgs.MsgCannotCancelBatchType)
+	}
+	if !id.Equals(fs.Flushing) {
+		return i18n.NewError(ctx, coremsgs.MsgCannotCancelBatchState)
```

Comment:
Maybe worth a separate error for this, including the ID of the one that _is_ being flushed (so the user can try and cancel that one instead)?